lib/shopify_api/bulk/query.ex

```diff
@@ -71,7 +71,8 @@ defmodule ShopifyAPI.Bulk.Query do
          {:ok, resp} <- ShopifyAPI.graphql_request(token, bulk_query, 10),
          :ok <- handle_errors(resp),
          bulk_query_id <- get_in(resp.response, ["bulkOperationRunQuery", "bulkOperation", "id"]),
-         {:ok, url} <- poll(token, bulk_query_id, opts[:polling_rate], opts[:max_poll_count]) do
+         {:ok, url} when is_binary(url) <-
+           poll(token, bulk_query_id, opts[:polling_rate], opts[:max_poll_count]) do
```

Comment:
🍎 I'd probably prefer the clarity of another else clause that explicitly handles the `{:ok, :no_objects}` response from `poll`,  as opposed  to it just being lumped in with the `error -> error` - no objects being returned isn't an error, and kind of obfuscates what's happening (took me a few to get it, hehe).

Reply:
Actually, this new `when is_binary` isn't necessary at all, is it?

Reply:
yea not a bad idea, I initially had one at some point in this code and it got lost :\ 
the other option is change it from "ok" to something like :error, :no_objects.... :thinking:

Reply:
Yeah, I'd argue returning 0 results isn't actually error, but ... :D

Reply:
It _is_ a bit wonky when the second element of that tuple is either a `url` string or a descriptive `atom`

Reply:
updated as per our conversation 👍